skopt/acquisition.py

```diff
@@ -264,3 +267,119 @@ def gaussian_ei(X, model, y_opt=0.0, xi=0.01, return_grad=False):
         return values, grad
 
     return values
+
+
+def gaussian_entropy_search(X, model, space, n_rep_points=50,
+                            n_trial_points=200, measure="EI", n_samples=500,
+                            random_state=None):
+    """
+    Uses entropy search to optimize the acquisition function.
+
+    1. It approximates the search space by a set of representer points sampled from
+       a suitable measure such as the "probability of improvement" or
+       "expected improvement".
+
+    2. p_min at a representer point X_i is calculated by the Integral
+       p(f(x)) \prod_{j=1}^N_rep \Theta(X_j - X_i) \\
+       Where P(f(x)) is the GP posterior and \Theta is the heaviside step function.
+       This can be used to calculate the base entropy.
+
+    3. Then for each candidate point, the change in the mean and the covariance
+       of each of these representer points can be computed. This in turn
+       can be used to compute the change in p_min for each of these representer
+       points.
+
+    Parameters
+    ----------
+    * `X` [array-like, shape=(n_samples, n_features)]:
+        Values where the acquisition function should be computed.
+
+    * `model` [sklearn estimator that implements predict with ``return_std``]:
+        The fit estimator that approximates the function through the
+        method ``predict``.
+        It should have a ``return_std`` parameter that returns the standard
+        deviation.
+
+    * `space` [Space instance]:
+        This is required to sample representer points.
+
+    * `n_rep_points` [int, default 50]:
+        Number of represter points.
+
+    * `n_trial_points` [int, default 200]:
+        Number of trial points sampled uniformly for each representer point
+        candidate.
+
+    * `n_samples` [int, default 500]:
+        Number of samples to sample from the GP posterior to approximate p_min
+
+    * `random_state` [int, RandomState instance, or None (default)]:
+        For reproducible results.
+    """
+    rng = np.random.RandomState(random_state)
+    n_features = X.shape[1]
+
+    # Step 1: Sample `n_representer_points` from the given nonmeasure.
+    # XXX: Use MCMC sampling to sample from the quadrature measure.
```

Comment:
If I understand it correctly, you resample the representer points for every call of `gaussian_entropy_search`. And `fmin_l_bfgs_b` calls it several times during the search for the next query point. This causes two issues:
 1) There is unnecessary computation   
 2) Since the representer points are different in each call, the values returned by the acquisition function are not compatible (e.g., evaluating the same point twice might return different values). This issue could be fixed by setting the same random seed before generating the representer points.